packages/mui-joy/src/Drawer/Drawer.tsx

```diff
@@ -0,0 +1,399 @@
+    '--unstable_popup-zIndex': `calc(${theme.vars.zIndex.modal} + 1)`,
+  },
+  position: 'fixed',
+  zIndex: theme.vars.zIndex.modal,
+  transitionProperty: 'visibility',
+  transitionDelay: ownerState.open ? '0s' : '300ms',
+  ...(!ownerState.open && {
+    visibility: 'hidden',
+  }),
+  [`& .${drawerClasses.backdrop}`]: {
+    opacity: ownerState.open ? 1 : 0,
+    transition: 'opacity 0.3s ease',
+  },
+}));
+
+const DrawerBackdrop = styled(ModalBackdrop as unknown as 'div', {
+  name: 'JoyDrawer',
+  slot: 'Backdrop',
+  overridesResolver: (props, styles) => styles.backdrop,
+})({});
+
+export const DrawerContent = styled(SheetRoot as unknown as 'div', {
+  name: 'JoyDrawer',
+  slot: 'Content',
+  overridesResolver: (props, styles) => styles.root,
+})<{ ownerState: DrawerOwnerState }>(({ theme, ownerState }) => ({
+  '--DrawerContent-radius': 0,
+  ...(ownerState.size === 'sm' && {
+    '--DrawerContent-padding': theme.spacing(2),
+    '--DrawerContent-gap': theme.spacing(0.75),
+    '--DrawerContent-titleOffset': theme.spacing(0.25),
+    '--DrawerContent-descriptionOffset': theme.spacing(0.25),
+    '--ModalClose-inset': theme.spacing(1.25),
+  }),
+  ...(ownerState.size === 'md' && {
+    '--DrawerContent-padding': theme.spacing(2.5),
+    '--DrawerContent-gap': theme.spacing(1.5),
+    '--DrawerContent-titleOffset': theme.spacing(0.25),
+    '--DrawerContent-descriptionOffset': theme.spacing(0.75),
+    '--ModalClose-inset': theme.spacing(1.5),
+  }),
+  ...(ownerState.size === 'lg' && {
+    '--DrawerContent-padding': theme.spacing(3),
+    '--DrawerContent-gap': theme.spacing(2),
+    '--DrawerContent-titleOffset': theme.spacing(0.5),
+    '--DrawerContent-descriptionOffset': theme.spacing(1),
+    '--ModalClose-inset': theme.spacing(1.5),
+  }),
+  boxShadow: theme.shadow.md,
+  borderRadius: 'var(--DrawerContent-radius)',
+  fontFamily: theme.vars.fontFamily.body,
+  lineHeight: theme.vars.lineHeight.md,
+  padding: 'var(--DrawerContent-padding)',
+  minWidth:
+    'min(calc(100vw - 2 * var(--DrawerContent-padding)), var(--DrawerContent-minWidth, 240px))',
+  outline: 0,
+  display: 'flex',
+  flexDirection: 'column',
+  position: 'fixed',
+  boxSizing: 'border-box',
+  overflow: 'auto',
+  ...(ownerState.anchor === 'left' && {
+    top: 0,
+    left: 0,
+    transform: ownerState.open ? 'translateX(0)' : 'translateX(-100%)',
+  }),
+  ...(ownerState.anchor === 'right' && {
+    top: 0,
+    right: 0,
+    transform: ownerState.open ? 'translateX(0)' : 'translateX(100%)',
+  }),
+  ...(ownerState.anchor === 'top' && {
+    top: 0,
+    transform: ownerState.open ? 'translateY(0)' : 'translateY(-100%)',
+  }),
+  ...(ownerState.anchor === 'bottom' && {
+    bottom: 0,
+    transform: ownerState.open ? 'translateY(0)' : 'translateY(100%)',
+  }),
+  height: ownerState.anchor!.match(/(left|right)/) ? '100%' : 'auto',
+  width: ownerState.anchor!.match(/(top|bottom)/) ? '100vw' : 'auto',
+  transition: 'transform 0.3s ease',
```

Comment:
tag @zanivan for reviewing the transition/animation.

Reply:
Just for clarification, you decided to remove the transitions of all components in favor of adding them in a "plugin" fashion way, but the Accordion and now the Drawer also have them.

Reply:
Just sharing my thoughts: Though I like the idea of having a more robust feature for animations and transitions, I think we can make some smaller additions meanwhile. This is mostly for the scenarios where a) it's very common to have animations, and b) for the newly introduced components. 
As for the existing components, we could potentially handle them on a case-by-case basis if they fall into category a), like `switch` for instance. However, if it gets overly complex, it might be better to hold off until the “animation plugin” is available.

Reply:
Update the transition looks better.

> Just for clarification, you decided to remove the transitions of all components in favor of adding them in a "plugin" fashion way, but the Accordion and now the Drawer also have them.

This is a bigger discussion, we should probably move this to the issue about it - https://github.com/mui/material-ui/issues/36695. In general in my opinion we should have a minimal transitions by default, that will make the components feel great - in addition we can have mechanisms for extending these defaults.

Reply:
@marcpachecog We will add a transition demo to give an example on how to customize it.